pkg/trace/agent/truncator_test.go

```diff
@@ -38,59 +38,82 @@ func testSpan() *pb.Span {
 }
 
 func TestTruncateResourcePassThru(t *testing.T) {
+	a := &Agent{conf: config.New()}
 	s := testSpan()
 	before := s.Resource
-	Truncate(s)
+	a.Truncate(s)
 	assert.Equal(t, before, s.Resource)
 }
 
 func TestTruncateLongResource(t *testing.T) {
+	a := &Agent{conf: config.New()}
 	s := testSpan()
 	s.Resource = strings.Repeat("TOOLONG", 5000)
-	Truncate(s)
+	a.Truncate(s)
 	assert.Equal(t, 5000, len(s.Resource))
 }
 
 func TestTruncateMetricsPassThru(t *testing.T) {
+	a := &Agent{conf: config.New()}
 	s := testSpan()
 	before := s.Metrics
-	Truncate(s)
+	a.Truncate(s)
 	assert.Equal(t, before, s.Metrics)
 }
 
 func TestTruncateMetricsKeyTooLong(t *testing.T) {
+	a := &Agent{conf: config.New()}
 	s := testSpan()
 	key := strings.Repeat("TOOLONG", 1000)
 	s.Metrics[key] = 42
-	Truncate(s)
+	a.Truncate(s)
 	for k := range s.Metrics {
-		assert.True(t, len(k) < traceutil.MaxMetricsKeyLen+4)
+		assert.True(t, len(k) < MaxMetricsKeyLen+4)
 	}
 }
 
 func TestTruncateMetaPassThru(t *testing.T) {
+	a := &Agent{conf: config.New()}
 	s := testSpan()
 	before := s.Meta
-	Truncate(s)
+	a.Truncate(s)
 	assert.Equal(t, before, s.Meta)
 }
 
 func TestTruncateMetaKeyTooLong(t *testing.T) {
+	a := &Agent{conf: config.New()}
 	s := testSpan()
 	key := strings.Repeat("TOOLONG", 1000)
 	s.Meta[key] = "foo"
-	Truncate(s)
+	a.Truncate(s)
 	for k := range s.Meta {
-		assert.True(t, len(k) < traceutil.MaxMetaKeyLen+4)
+		assert.True(t, len(k) < MaxMetaKeyLen+4)
 	}
 }
 
 func TestTruncateMetaValueTooLong(t *testing.T) {
+	a := &Agent{conf: config.New()}
 	s := testSpan()
 	val := strings.Repeat("TOOLONG", 25000)
 	s.Meta["foo"] = val
-	Truncate(s)
+	a.Truncate(s)
 	for _, v := range s.Meta {
-		assert.True(t, len(v) < traceutil.MaxMetaValLen+4)
+		assert.True(t, len(v) < MaxMetaValLen+4)
 	}
 }
+
+func TestTruncateResource(t *testing.T) {
+	a := &Agent{conf: config.New()}
+	t.Run("over", func(t *testing.T) {
+		r, ok := a.TruncateResource("resource")
+		assert.True(t, ok)
+		assert.Equal(t, "resource", r)
+	})
+
+	t.Run("under", func(t *testing.T) {
```

Comment:
I think the `under` test is actually an "equal" test since it just uses the max value :P (Note these tests added here were simply pulled from the traceutil package when the `TruncateResource` function was pulled here)